### PR TITLE
Set search domain on regular containers too

### PR DESCRIPTION
**Pull request by alena1108**
Also added authoritative section to default dns settings

### CODE REVIEW

File: code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreCreate.java [hunk 1]
```diff
@@ -33,12 +36,36 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
...
         Map<String, Object> labels = DataAccessor.fieldMap(instance, InstanceConstants.FIELD_LABELS);
+        Map<Object, Object> data = new HashMap<>();
         if (labels.containsKey(SystemLabels.LABEL_AGENT_CREATE)
                 && labels.get(SystemLabels.LABEL_AGENT_CREATE).equals("true")) {
-            Map<Object, Object> data = new HashMap<>();
             List<String> dataVolumes = DataAccessor.fieldStringList(instance, InstanceConstants.FIELD_DATA_VOLUMES);
             dataVolumes.add(AgentConstants.AGENT_INSTANCE_BIND_MOUNT);
             data.put(InstanceConstants.FIELD_DATA_VOLUMES, dataVolumes);
+        }
+        if (InstanceConstants.KIND_CONTAINER.equalsIgnoreCase(instance.getKind())) {
```
Comment (ibuildthecloud) [inline code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreCreate.java]: Do `InstanceContants.CONTAINER_LIKE.contains(instance.getKind())`

File: tests/integration/cattletest/core/test_docker.py [hunk 2]
```diff
@@ -637,9 +637,9 @@ def test_container_fields(docker_client, super_client):
     assert set(c.data['dockerInspect']['HostConfig']['CapAdd']) == set(caps)
     assert set(c.data['dockerInspect']['HostConfig']['CapDrop']) == set(caps)
     actual_dns = c.data['dockerInspect']['HostConfig']['Dns']
-    assert set(actual_dns) == set(['8.8.8.8', '1.2.3.4'])
+    assert set(actual_dns) == set(['8.8.8.8', '1.2.3.4', '169.254.169.250'])
```
Comment (ibuildthecloud) [inline tests/integration/cattletest/core/test_docker.py]: Why did this change?  I though if the user configures DNS we don't override it?

File: code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreCreate.java (hunk 1 above)
Comment (alena1108) [inline code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreCreate.java]: ok

File: tests/integration/cattletest/core/test_docker.py (hunk 2 above)
Comment (alena1108) [inline tests/integration/cattletest/core/test_docker.py]: no...we still put our dns there, and its set on the container, and our dns will be used. We just don't propagate it to dns config yml

**Comment by alena1108** [comment]
Also fixed the order of search domains in dns config

**Comment by alena1108** [comment]
Here is how default section will look like:

```
"default": {
		"search": ["rancher.internal"],
		"recurse": ["192.168.1.1"], // this is read from host config
		"authoritative": ["rancher.internal"],
....
```

client section:

```
"10.42.114.203": {
		"search": ["foo.rancher.internal", "rancher.internal"],
		"recurse": [],
		"authoritative": [],
```

/etc/resolv.conf:

```
search foo.rancher.internal rancher.internal
nameserver 169.254.169.250
```

@vincent99 @ibuildthecloud confirm that everything above is correct

File: tests/integration/cattletest/core/test_docker.py (hunk 2 above)
Comment (ibuildthecloud) [inline tests/integration/cattletest/core/test_docker.py]: omg, i can't keep all this straight

**Comment by ibuildthecloud** [comment]
@alena1108 do you define `rancher-metadata.rancher.internal => 169.254.169.250`in the defaults section?

**Comment by alena1108** [comment]
@ibuildthecloud yes I do

File: tests/integration/cattletest/core/test_docker.py (hunk 2 above)
Comment (alena1108) [inline tests/integration/cattletest/core/test_docker.py]: lol :)

**Pull request merged**